pytorch_to_returnn/converter/converter.py

```diff
@@ -156,6 +157,27 @@ def _make_tf_feed_dict(self, input: Data):
     else:
       for i, size in input.size_placeholder.items():
         d[size] = [self._inputs_np.shape[input.get_batch_axis(i)]] * n_batch  # not so relevant
+    if out_entry:
+      visited_entries = set()
+
+      def _get_non_deterministic_tensor_entries(entry_: TensorEntry):
```

Comment:
Note, actually I would not introduce such a function and do this recursively at all. Recursion is somewhat inefficient. Rather, I would start defining a queue, like:
```python
entries_to_visit = [out_entry]
```

And then you have a loop:
```python
while entries_to_visit:
  entry_ = entries_to_visit.pop(0)
  ...
```
And then the code from the function follows.
You would replace the recursive call by:
```python
entries_to_visit.append(prev_entry_)
```